### PR TITLE
Collection as Set

### PR DESCRIPTION
**Pull request by g4s8**
Introduced `CollectionAsSet` instead of `FixedSet` in #22

### CODE REVIEW

**Comment by 0crat** [comment]
@g4s8 I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

File: src/main/java/org/cactoos/list/CollectionAsSet.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    private final Collection<T> collection;
+
+    /**
+     * Ctor.
+     *
+     * @param collection Source collection
+     */
+    public CollectionAsSet(final Collection<T> collection) {
+        super();
```
Comment (englishman) [inline src/main/java/org/cactoos/list/CollectionAsSet.java]: Is it needed?

File: src/test/java/org/cactoos/list/CollectionAsSetTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+     * Test set contains element.
+     */
+    @Test
+    public void containsTest() {
+        final int element = 6324;
+        MatcherAssert.assertThat(
+            new CollectionAsSet<>(
+                new IterableAsList<>(
+                    // @checkstyle MagicNumber (1 line)
+                    new ArrayAsIterable<>(1, 2, element, 4)
```
Comment (englishman) [inline src/test/java/org/cactoos/list/CollectionAsSetTest.java]: 0, 1 and 2 can be used here
http://checkstyle.sourceforge.net/config_coding.html#MagicNumber
So `// @checkstyle MagicNumber (1 line)` isn't needed at all.

File: README.md [hunk 3]
```diff
@@ -200,6 +200,10 @@ mvn clean install -Pqulice
 Note: [Checkstyle](https://en.wikipedia.org/wiki/Checkstyle) is used as a static code analyze tool with
 [checks list](http://checkstyle.sourceforge.net/checks.html) in GitHub precommits.
 
+## Contributors
+
+ * [Kirill Che. g4s8.public@gmail.com](https://github.com/g4s8)
```
Comment (englishman) [inline README.md]: :+1:

File: src/main/java/org/cactoos/list/CollectionAsSet.java (hunk 1 above)
Comment (g4s8) [inline src/main/java/org/cactoos/list/CollectionAsSet.java]: @englishman PMD aked me to call super here. see `CallSuperInConstructor` in http://pmd.sourceforge.net/pmd-4.3.0/rules/index.html

File: src/test/java/org/cactoos/list/CollectionAsSetTest.java (hunk 2 above)
Comment (g4s8) [inline src/test/java/org/cactoos/list/CollectionAsSetTest.java]: @englishman thanks, removed

File: src/main/java/org/cactoos/list/CollectionAsSet.java (hunk 1 above)
Comment (englishman) [inline src/main/java/org/cactoos/list/CollectionAsSet.java]: OMG... do we have also PMD checks her (not only Checkstyle)?

Comment (g4s8) [inline src/main/java/org/cactoos/list/CollectionAsSet.java]: @englishman yes, all checks in one plugin https://github.com/teamed/qulice

File: src/main/java/org/cactoos/list/CollectionAsSet.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Collection;
+import java.util.Iterator;
+
+/**
+ * Collection as set.
+ *
+ * @author Kirill (g4s8.public@gmail.com)
+ * @version $Id$
+ * @param <T> Elements type
+ * @since 0.1
```
Comment (yegor256) [inline src/main/java/org/cactoos/list/CollectionAsSet.java]: @g4s8 0.1 is already out. Put `0.2` here

File: src/test/java/org/cactoos/list/CollectionAsSetTest.java [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    @Test
+    public void sizeTest() {
+        final int size = 412;
+        MatcherAssert.assertThat(
+            new CollectionAsSet<>(
+                new IterableAsList<>(
+                    new Repeat<>(0, size)
+                )
+            ),
+            Matchers.hasSize(size)
```
Comment (yegor256) [inline src/test/java/org/cactoos/list/CollectionAsSetTest.java]: @g4s8 hm... as far as I understand what a `Set` is, the size should be `1` :) A set may not have duplicates.

**Comment by yegor256** [comment]
@g4s8 see my comments above

File: src/test/java/org/cactoos/list/CollectionAsSetTest.java (hunk 5 above)
Comment (g4s8) [inline src/test/java/org/cactoos/list/CollectionAsSetTest.java]: @yegor256 fixed. Don't know how I wrote such test)

**Comment by g4s8** [comment]
@yegor256 fixed

**Comment by g4s8** [comment]
@yegor256 ok, you're right

**Pull request closed without merging**